src/classes/UTIL_AbstractCommon_BATCH.cls

```diff
@@ -0,0 +1,521 @@
+* over.  The main limitation of this implementation is that it will not be possible to
+* provide an order to the records returned because it has to execute in a specific order
+* when running in LDV Chunking Mode.
+*
+* Due to the fact that constructors are not inherited by default, any constructors that
+* are desired need to be created using the signature as desired, and then in the constructor
+* body providing a super method and any params, e.g. super(batchSize);
+*
+* Where clause can be added by overriding getWhereClause() method
+*
+* Additional Where Clause can be configured by passing to constructor
+*
+* If Additional Where Clause is added, it can be validated by executing validateQuery().  If
+* it returns null, it is syntactically correct.  If it returns an Exception object, something about the
+* query is invalid.  The Additional Where and/or Where clause will be applied to the inner query for
+* each batch, if executing in LDV Chunking Mode, and the main batch query if executing in non LDV
+* Chunking mode.
+*
+* Future: Add exception handling, and logging support so implementing clases can inherit.
+* Recommend registering a default handler using an interface that can be switched out/overriden
+* as desired for specific use cases.
+*/
+public abstract inherited sharing class UTIL_AbstractCommon_BATCH implements Database.Batchable<SObject>, Database.Stateful {
+
+    /**
+     * @description The Maximum LDV Chunk size that is allowed to be used.  If > MAX will use the MAX
+     */
+    @testVisible private static final Integer MAX_LDV_CHUNK_SIZE = 10000000;
+    /**
+     * @description Default size to use when chunking records
+     */
+    @testVisible private static final Integer DEFAULT_LDV_CHUNK_SIZE = 1000000;
+    /**
+     * @description Default Maximum Size in # of records between min and max Id before it must run in LDV Chunking Mode
+     */
+    @testVisible private static final Integer DEFAULT_MAX_NON_LDV_SIZE = 1000000;
+    /**
+     * @description Default size of the batch when re-executing in chunk mode, if not provided
+     */
+    @testVisible private static final Integer DEFAULT_BATCH_SIZE = 500;
+
+    @testVisible private Id startId;
+    @testVisible private Id endId;
+    @testVisible private Id currentEndId;
+    @testVisible private Boolean endReached;
+    @testVisible private Integer batchSize = DEFAULT_BATCH_SIZE;
+    @testVisible private Integer chunkIteration;
+    @testVisible private Boolean ldvMode = null;
+    @testVisible private String additionalWhereClause;
+
+    /**
+     * @description No argument constructor that assumes default batch size of 500, and that ldvMode will be
+     * determined by record distance between min and max Id for main object of batch
+     */
+    public UTIL_AbstractCommon_BATCH() {
+        // batchSize, ldvMode, additionalWhereClause
+        this(DEFAULT_BATCH_SIZE, false,'');
```

Comment:
Does the `additionalWhereClause` value have to be passed as an empty string instead of a null?